src/core/src/main/kotlin/org/apache/jmeter/threads/openmodel/OpenModelThreadGroup.kt

```diff
@@ -204,7 +204,7 @@ public class OpenModelThreadGroup :
             val seed = randomSeed
             val rnd = if (seed == 0L) Random() else Random(seed)
             val gen = ThreadScheduleProcessGenerator(rnd, parsedSchedule)
-            val testStartTime = JMeterContextService.getTestStartTime()
+            val testStartTime = System.currentTimeMillis()
```

Comment:
Do we have something like "setup completed timestamp" or "main test started (excluding setup)"?

Reply:
I am not aware of such a thing. 

But do you think this is needed? ThreadGroup#start gets called, when the thread group should start, so this seems to enough (for now?).

Reply:
`JMeterContextService.getTestStartTime` is used in `ConstantThroughputTimer`, `PreciseThroughputTimer`.
I expect the classes will have the same issue regarding long setup thread groups.
At the same time, all of those might appear under setup thread groups as well. WDYT of using something like "thread group start time"?

Of course, for `OMTG` it is a bit easier as `start` corresponds to the thread group start, however, the timers should probably refer to the start time of the corresponding thread groups.

Reply:
Right, if they use the same timestamp to calculate the beginning of the thread group, they are misled, too. 

Do the timers have access to their respective thread groups?

Reply:
There's `JMeterContextService.getContext().getThreadGroup()`.
However, `AbstractThreadGroup` does not track its start time :-/

Reply:
It sounds like a good plan.

`ConstantThroughputTimer` has "shared" mode which means it somehow operates across the thread groups, so I am not sure which "start" timestamp it should use then.

Reply:
Yes, there has to be done more logic in the `ConstantThrouputTimer`. The delay has to be calculated depending on the mode. So I left it out now, but it could be added later.